### PR TITLE
Document and rename parameter example test

### PR DESCRIPTION
**Pull request by eivindjahren**
Small cleanup in typing, naming and documentation of the parameter_example test.

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/ert/ui_tests/cli/test_parameter_passing.py [hunk 1]
```diff
@@ -1,16 +1,25 @@
+from __future__ import annotations
+
+"""
+The following test generates different kind of parameters
```
Comment (yngve-sk) [inline tests/ert/ui_tests/cli/test_parameter_passing.py:4]: nitpick: kindS

File: tests/ert/ui_tests/cli/test_parameter_passing.py [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
+and places their configuration in the config file. Then a
```
Comment (yngve-sk) [inline tests/ert/ui_tests/cli/test_parameter_passing.py:5]: aN ensemble experiment

File: tests/ert/ui_tests/cli/test_parameter_passing.py [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
+ensemble experiment is ran and we assert that each realization
+got the parameters correctly in the runpath.
```
Comment (yngve-sk) [inline tests/ert/ui_tests/cli/test_parameter_passing.py:7]: nitpick: "each realization got .." reads like slang

**Review by yngve-sk — APPROVED** [review]
LGTM, some grammar nitpicks on comments

File: tests/ert/ui_tests/cli/test_parameter_passing.py (hunk 3 above)
Comment (eivindjahren) [reply tests/ert/ui_tests/cli/test_parameter_passing.py:7]: Fixed

File: tests/ert/ui_tests/cli/test_parameter_passing.py (hunk 2 above)
Comment (eivindjahren) [reply tests/ert/ui_tests/cli/test_parameter_passing.py:5]: Fixed

**Pull request merged**